fluent_contents/forms/widgets.py

```diff
@@ -49,11 +50,14 @@ def render(self, name, value, attrs=None):
         """
         other_instance_languages = None
         if value and value != "-DUMMY-":
-            parent = Placeholder.objects.get(pk=int(value)).parent
-            language_code = get_parent_language_code(parent)
-            if language_code:
+            placeholder = Placeholder.objects.get(pk=int(value))
+
+            parent = placeholder.parent.__class__.objects.language(
+                self.language_code).get(pk=placeholder.parent_id)
```

Comment:
Instead of refetching it, you can use `parent.set_current_language(language_code)`.

I must say I'm very impressed with the way to manage to pass the current language code through all objects.
This also makes me reconsider this old hack.
It fetches `parent` and that's the same object as `obj` in `formfield_for_dbfield()`.

I think it would make more sense to either:

* pass  `parent_object` directly to the `PlaceholderField`
* pass the language choices to the `PlaceholderField`

What do you think is the best solution?